pkg/framework/context/context.go

```diff
@@ -129,13 +129,19 @@ func (p *Provider) InboundMessageHandler() transport.InboundMessageHandler {
 		// in case of no services are registered for given message type,
 		// find generic inbound services registered for given message header
 		for _, svc := range p.msgSvcProvider.Services() {
-			if svc.Accept(msg.Header) {
+			h := &service.Header{}
+			err = msg.ToStruct(h)
+			if err != nil {
+				return err
+			}
+
+			if svc.Accept(h) {
```

Comment:
Passing whole message may be risky to check acceptance criteria, for now add a TODO to pass subset of DIDComm msg map.

Reply:
Accept may use an interface e.g
```
type Typer interface {
   Type() string
}
```

Reply:
We are also going to check it based on `~purpose`.

Reply:
@Moopli ^^^